src/main/java/com/squareup/javapoet/CodeWriter.java

```diff
@@ -247,6 +265,27 @@ public CodeWriter emit(CodeBlock codeBlock) throws IOException {
           break;
 
         default:
+          if (potentiallyStaticImported == null) {
+            emitAndIndent(part);
+            break;
+          }
+          ClassName pendingClassName = potentiallyStaticImported;
+          potentiallyStaticImported = null;
+          if (part.startsWith(".")) {
+              int endIndex = part.indexOf('(');
+              if (endIndex < 0) endIndex = part.length();
+              String memberName = part.substring(1, endIndex);
+              checkState(isIdentifier(memberName), "identifier expected, but got: %s", memberName);
+              String candidate = pendingClassName.canonicalName + "." + memberName;
+              String wildcard = pendingClassName.canonicalName + ".*";
+              if (staticImportSignatures.contains(candidate)
+                  || staticImportSignatures.contains(wildcard)) {
+                String partTail = part.substring(memberName.length() + 1);
```

Comment:
Instead of this line and 284, isn't it just, `emitAndIndent(part.substring(1));` ?